src/objects.js

```diff
@@ -1,38 +1,71 @@
 // Complete the following underscore functions.
 // Reference http://underscorejs.org/ for examples.
+const arrayMethods = require('../src/arrays');
 
 const keys = (obj) => {
   // Retrieve all the names of the object's properties.
   // Return the keys as strings in an array.
   // Based on http://underscorejs.org/#keys
+
+  // not sure how to do this without just using an Object method
+  return arrayMethods.reduce(Object.entries(obj), (array, [key, value]) => {
+    array.push(key);
+    return array;
+  }, []);
 };
 
 const values = (obj) => {
   // Return all of the values of the object's own properties.
   // Ignore functions
   // http://underscorejs.org/#values
+  return arrayMethods.reduce(Object.entries(obj), (array, [key, value]) => {
+    array.push(value);
+    return array;
+  }, []);
 };
 
 const mapObject = (obj, cb) => {
   // Like map for arrays, but for objects. Transform the value of each property in turn.
   // http://underscorejs.org/#mapObject
+  const newObj = {};
+  for (let i = 0; i < keys(obj).length; i++) {
```

Comment:
Awesome. Passing your object back to your very one 'key' function. This is good stuff.